src/ununpack/agent/checksum.c

```diff
@@ -183,34 +185,39 @@ Cksum *	SumComputeFile	(FILE *Fin)
  */
 Cksum *	SumComputeBuff	(CksumFile *CF)
 {
-  int rc;
-  SHA1Context sha1;
-  MyMD5_CTX md5;
   Cksum *Sum;
+  gcry_md_hd_t checksumhandler;
+  gcry_error_t checksumError;
+  unsigned char *tempBuff;
 
-  Sum = (Cksum *)calloc(1,sizeof(Cksum));
-  if (!Sum) return(NULL);
-  Sum->DataLen = CF->MmapSize;
-
-  MyMD5_Init(&md5);
-  rc = SHA1Reset(&sha1);
-  if (rc)
+  checksumError = gcry_md_open(&checksumhandler, GCRY_MD_NONE, 0);
+  if (! checksumhandler)
   {
-    LOG_ERROR("Unable to initialize sha1\n");
-    free(Sum);
+    LOG_ERROR("Unable to initialize checksum\n");
     return(NULL);
   }
-
-  MyMD5_Update(&md5,CF->Mmap,CF->MmapSize);
-  MyMD5_Final(Sum->MD5digest,&md5);
-  SHA1Input(&sha1,CF->Mmap,CF->MmapSize);
-  rc = SHA1Result(&sha1,Sum->SHA1digest);
-  if (rc)
+  checksumError = gcry_md_enable(checksumhandler, GCRY_MD_MD5);
```

Comment:
check for the error code here as well
if (gcry_err_code(checksumError) != GPG_ERR_NO_ERROR)

Reply:
Done, please check.